data/org.freedesktop.portal.DynamicLauncher.xml

```diff
@@ -0,0 +1,301 @@
+        The checksum at the end of the file name is an implementation detail in
+        Epiphany and not required by the portal.
+
+        If the calling process is unsandboxed, @desktop_file_id need only have
+        a ".desktop" suffix.
+
+        This method is intended to be called by the application that created the
+        launcher, e.g. a web browser, so it can clean up associated data such as
+        the profile directory as part of the uninstallation. Consequently, the
+        proper way for a software center to remove a launcher is by using the APIs
+        provided by the application that installed it. For example, for GNOME
+        Software to remove web launchers created by Epiphany, it would use the
+        org.gnome.Epiphany.WebAppProvider D-Bus interface.
+
+        Please note that this method call will fail if the specified launcher
+        already does not exist.
+
+        The @options vardict currently has no supported entries.
+    -->
+    <method name="Uninstall">
+      <arg type="s" name="desktop_file_id" direction="in"/>
+      <arg type="a{sv}" name="options" direction="in"/>
+    </method>
+    <!--
+        GetDesktopEntry:
+        @desktop_file_id: The .desktop file name
+        @contents: the contents of the named .desktop file
+
+        This function returns the contents of a desktop file with the name
+        @desktop_file_id in @contents. If the calling process is sandboxed,
+        @desktop_file_id must have the app ID followed by a period as a prefix;
+        see the Uninstall() method documentation.
+
+        If the calling process is unsandboxed, @desktop_file_id need only have
+        a ".desktop" suffix.
+
+        This method only works for desktop files that were created by the
+        dynamic launcher portal.
+    -->
+    <method name="GetDesktopEntry">
+      <arg type="s" name="desktop_file_id" direction="in"/>
+      <arg type="s" name="contents" direction="out"/>
+    </method>
+    <!--
+        Launch:
+        @desktop_file_id: The .desktop file name
+        @options: Vardict with optional further onformation
+
+        This function launches the app specified by @desktop_file_id. If the
+        calling process is sandboxed, @desktop_file_id must have the app ID
+        followed by a period as a prefix; see the Uninstall() method
+        documentation.
+
+        If the calling process is unsandboxed, @desktop_file_id need only have
+        a ".desktop" suffix.
+
+        This method only works for desktop files that were created by the
+        dynamic launcher portal.
+
+        Supported keys in the @options vardict include:
+        <variablelist>
+          <varlistentry>
+            <term>activation_token s</term>
+            <listitem><para>
+              A token that can be used to activate the chosen application.
+            </para><para>
+              The activation_token option was introduced in version 4 of the interface.
+            </para></listitem>
```

Comment:
This new portal certainly won't hit the ground running like this :slightly_smiling_face: